### DISCUSSION TITLE
Media Library and SFTP

### DISCUSSION THREAD

**Discussion by milosh-96** [OrchardCMS/OrchardCore#11900] [Q&A]
Hello,

I'm working with Orchard Core CMS and I would like to have a centralized media storage (I hope it's the right term) so the uploaded files (mostly images) aren't stored in the local App_Data but in a specified folder on the remote server. Later, the files would be served via "https://uploads.site.com". 

Basically, I want to achieve something like this, but with on my own server - with files.

> The feature replaces the default App_Data file based media store with an Azure Media Storage Provider.

How should I approach this, any tips or links? 

(I saw S3 Media module but I'm still not sure how to convert it to SFTP).

**Comment by hishamco** [comment]
> but in a specified folder on the remote server

Do you mean the same server where `App_Data` exists? If Yes, ~~simply you can customize the `FileSystemStore`~~. If No, you need to create another mesdia storage provider

**Reply by hishamco** [reply]
In the first case you just need to set AssetsPath, from `MediaOptions`

**Reply by milosh-96** [reply]
> > but in a specified folder on the remote server
> 
> Do you mean the same server where `App_Data` exists? If Yes, simply you can customize the `FileSystemStore`. If No, you need to create another mesdia storage provider

I think I will need a storage provider.
I drew this graph that will illustrate my idea.
![image](https://user-images.githubusercontent.com/13017737/175444879-d29cbd04-5005-4035-b1b2-f3f4665467e3.png)

**Comment by hishamco** [comment]
Yep, you need to create an `FtpMediaStorageProvider`
